src/cdsetool/query.py

```diff
@@ -324,3 +324,25 @@ def _get_describe_doc(
             _describe_docs[collection] = res.content
             return res.content
     assert False, f"Failed {attempts} times to get collection {collection}, giving up."
+
+
+def get_odata_by_name(name: str, proxies: Union[Dict[str, str], None] = None) -> Dict:
+    """Get odata for checksum given a product name"""
+    session = Credentials.make_session(None, False, Credentials.RETRIES, proxies)
+    attempts = 0
+    while attempts < 10:
+        attempts += 1
+        with session.get(
+            "https://catalogue.dataspace.copernicus.eu"
+            f"/odata/v1/Products?$filter=Name eq '{name}'"
+        ) as res:
+            if res.status_code >= 500:
```

Comment:
They actually return 404 every now and then as well, and then a bit later the system is fine and the same request works. So I think the check needs to be != 200 here as well.